wai-aria/scripts/aria-utils.js

```diff
@@ -192,5 +192,63 @@ const AriaUtils = {
     this.verifyLabelsBySelector(".ex-label-only", labelTestNamePrefix);
     this.verifyRolesBySelector(".ex-role-only", roleTestNamePrefix);
   },
-};
 
+
+  /*
+  Verifies that the subtree for a given accessible node matches the specified
+  tree structure.
+  This takes either a DOM Element or a CSS selector. It wraps the call in
+  promise_test.
+  For example:
+  <div id="listbox" role="listbox" aria-label="listbox">
+    <div id="option1" role="option" aria-label="option1"></div>
+    <div id="option2" role="option" aria-label="option2"></div>
+  </div>
+  ...
+  AriaUtils.verifyAccessibilitySubtree("#listbox", {
+    role: "listbox",
+    label: "listbox",
+    children: [
+      { role: "option", label: "option1", children: [] },
+      { role: "option", label: "option2", children: [] },
+    ],
+  });
+  */
+  verifyAccessibilitySubtree: function(subtreeRoot, expectedTree) {
+    const desc = subtreeRoot;
+    if (!(subtreeRoot instanceof Element)) {
+      subtreeRoot = document.querySelector(subtreeRoot);
+      if (!subtreeRoot) {
+        throw `selector passed to verifyAccessibilityTree("${subtreeRoot}") doesn't match an element`;
+      }
+    }
+    promise_test(async t => {
+      const acc = await test_driver.get_element_accessible_properties(subtreeRoot);
+      await AriaUtils._assertAccessibilitySubtree(acc, expectedTree, desc);
+    }, `accessibility tree for ${desc}`);
+  },
+
+
+  _assertAccessibilitySubtree: async function(accProps, expectedTree, position) {
```

Comment:
Even though this is an implementation detail, it might be nice to have a comment explaining the functionality and arguments for the benefit of future archaeologists/maintainers.